go-sms_inbound.js

```diff
@@ -236,29 +240,42 @@ go.utils = {
             });
     },
 
-    // Create a new contact with the minimum required details
-    create_contact: function(msisdn, im) {
-        var payload = {
-            "details": {
+    // Create a new identity
+    create_identity: function(im, msisdn, communicate_through_id, operator_id) {
+        var payload = {};
+
+        // compile base payload
+        if (msisdn) {
+            payload.details = {
```

Comment:
Just put the content of get_addresses in line here, it's not adding DRYness, just unnecessary jumping